master/internal/api_generic_tasks.go

```diff
@@ -148,7 +148,7 @@ func (a *apiServer) getGenericTaskLaunchParameters(
 	taskSpec.UserSessionToken = token
 
 	genericTaskSpec.Base = taskSpec
-	genericTaskSpec.GenericTaskConfig = taskConfig
+	genericTaskSpec.GenericTaskConfig = &taskConfig
```

Comment:
this is a metacomment but do you wanna make two PRs here?

One for the test failure (we think it is caused by not waiting for some tasks)

And a second PR (with tests) for the job queue not being restored and priority set?

Reply:
Removed the job queue stuff from PR. Everything in this PR should make the tests pass